packages/render-fidelity-tools/src/components/renderers/babylon-viewer.ts

```diff
@@ -0,0 +1,151 @@
+/* @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License atQ
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '@babylonjs/loaders/glTF';
+import {ArcRotateCamera, Engine, HDRCubeTexture, Scene, SceneLoader, Vector3} from '@babylonjs/core';
+import {css, customElement, html, LitElement, property} from 'lit-element';
+
+import {ScenarioConfig} from '../../common.js';
+
+// const IS_BINARY_RE = /\.glb$/;
+
+const $initialize = Symbol('initialize');
+const $updateScenario = Symbol('scenario');
+const $updateSize = Symbol('updateSize');
+const $render = Symbol('render');
+const $canvas = Symbol('canvas');
+const $engine = Symbol('engine');
+const $scene = Symbol('scene');
+const $camera = Symbol('camera');
+const $hdrTexture = Symbol('hdrTexture');
+
+
+@customElement('babylon-viewer')
+export class BabylonViewer extends LitElement {
+  @property({type: Object}) scenario: ScenarioConfig|null = null;
+  private[$canvas]: HTMLCanvasElement|null;
+  private[$engine]: Engine;
+  private[$scene]: Scene;
+  private[$camera]: ArcRotateCamera;
+  private[$hdrTexture]: HDRCubeTexture;
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+  }
+
+  updated(changedProperties: Map<string, any>) {
+    super.updated(changedProperties);
+    this[$updateSize]();
+
+    if (changedProperties.has('scenario') && this.scenario != null) {
+      this[$updateScenario](this.scenario);
+    }
+  }
+
+  static get styles() {
+    return css`
+:host {
+ display: block;
+}
+`;
+  }
+
+  render() {
+    this[$render]();
+    return html`<canvas id="canvas"></canvas>`;
+  }
+
+
+  private[$initialize]() {
+    this[$canvas] = this.shadowRoot!.querySelector('canvas');
+    this[$engine] = new Engine(this[$canvas], true);
+    this[$scene] = new Scene(this[$engine]);
+  }
+
+
+  private async[$updateScenario](scenario: ScenarioConfig) {
+    if (this[$scene] != null) {
+      this[$scene].dispose();
```

Comment:
Scene require disposal because if not, when the scenario changes, the former scenario's model is still in the scene. In the babylonjs's forum I found this is the recommended way to clear all models and meshes. 
Engine seems to be independent of scene. it doesn't need to be disposal. I think in babylonjs everything is represented as tree node, so the engine is the root node and the scene is a child node of engine.
I think environment texture should be disposal because some scenario has different environment texture.